### PR TITLE
[Product AI v2 M1] New preview screen

### LINKED ISSUE #11927: [Product AI v2 M1] Create a new preview screen for the new milestone

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #11927 

### Description
This PR adds a new screen showing the product preview during the AI creation.

The PR is quite big, but the review should be simple I hope, the cause of the large changes is the file `AiProductPreviewScreen`, and its content was mostly copied from the old implementation in `ProductPreviewSubScreen`, so it can be skipped.

### Testing information
1. Open the app and make sure you are signed using a Jetpack website.
2. Open product list.
3. Tap on Add a new product.
4. Select "Create a product with AI"
5. Enter any text on the field.
6. Click on "Generate Product Details"
7. Confirm the preview screen is shown with some sample data.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          7 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>LocalImage</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>AIProductModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>Shipping</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>AiProductPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>Success</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/Image.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.woocommerce.android.model
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+sealed interface Image : Parcelable {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/Image.kt:6]: This is the abstraction we discussed p1720520281796849/1720520261.479679-slack-C03L1NF1EA3, please share any thoughts you have on the implementation.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+import com.woocommerce.android.model.Product
+import com.woocommerce.android.model.ProductCategory
+import com.woocommerce.android.model.ProductTag
+import com.woocommerce.android.ui.products.ProductHelper
+import com.woocommerce.android.ui.products.ProductStatus.DRAFT
+import com.woocommerce.android.ui.products.ProductType.SIMPLE
+import java.math.BigDecimal
+
+data class AIProductModel(
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt:11]: Before, our implementation was using a regular `Product` for representing the AI model, where the JSON parsing was an implementation detail hidden inside `GenerateProductWithAI`, but now our model will need to provide different versions of some fields, so I introduced this new model which we'll use for this on next steps.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    @Suppress("UNUSED_PARAMETER")
+    fun onFeedbackReceived(positive: Boolean) {
+        TODO()
+    }
+
+    fun onBackButtonClick() {
+        triggerEvent(MultiLiveEvent.Event.Exit)
+    }
+
+    sealed interface State {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewViewModel.kt:62]: The `State` was copied from the previous implementation in `ProductPreviewSubViewModel`

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/productinfo/AiProductPromptViewModel.kt [hunk 4]
```diff
@@ -71,22 +72,27 @@ class AiProductPromptViewModel @Inject constructor(
...
+        )
     }
 
     fun onToneSelected(tone: Tone) {
         _state.value = _state.value.copy(selectedTone = tone)
     }
 
-    fun onMediaSelected(mediaUri: String) {
-        _state.value = _state.value.copy(mediaUri = mediaUri)
+    fun onMediaSelected(image: Image) {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/productinfo/AiProductPromptViewModel.kt:91]: I updated the implementation here to use the new `Image` model.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/Image.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/model/Image.kt:6]: Looks good to me 👍🏼

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt (hunk 2 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt:11]: Is there any reason for not adding the new `Image` model to this `AIProductModel` ?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewScreen.kt [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+                        ),
+                        AiProductPreviewViewModel.ProductPropertyCard(
+                            icon = R.drawable.ic_gridicons_list_checkmark,
+                            title = R.string.product_inventory,
+                            content = "In stock"
+                        )
+                    )
+                )
+            ),
+            onFeedbackReceived = {},
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewScreen.kt]: Minor but this is missing `selectedVariant` in order for the code to compile. 

```suggestion
                    )
                ),
                selectedVariant = 0,
            ),
            onFeedbackReceived = {},
```

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Nice job, @hichamboushaba, I just left a couple of minor suggestions or questions. Let me know what you think.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductModel.kt:11]: Yes, the `AIProductModel` will be parsed from the JSON we get from the AI response, whereas the image is supplied by the user directly.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewScreen.kt (hunk 5 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewScreen.kt]: Sorry about this, I made some last minute changes, and this was a left over.

**Pull request merged**